doc/api/punycode.md

```diff
@@ -1,6 +1,11 @@
 # punycode
 
-    Stability: 2 - Stable
+    Stability: 0 - Deprecated
+
+**The version of the punycode module bundled in Node.js is being deprecated**.
+In a future major version of Node.js this module will be removed. Users
```

Comment:
Hard deprecations are a bit touchy, since it means potentially breaking users' code. It is possible that some new data comes up between v7.x and v8.x versions that would make the CTC consider a hard deprecation later than v8.x releases.

If mentioning a specific version is considered to be a good incentive for users to move to using the npm module, then I agree we should do that. Otherwise, it seems like it would not be that useful for users, and it would actually limit our options.

So my question is: what are we trying to achieve by mentioning a specific version?